File: 125-valid-palindrome/valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        n = []
        for i in range(len(s)-1,-1,-1):
            if s[i].isalnum():
                n.append(s[i].lower())
        n = "".join(n)
        return n==n[::-1]
        

File: 125-valid-palindrome/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        StringBuilder str = new StringBuilder();
        for(int i=s.length()-1;i>=0;i--){
            if (Character.isLetterOrDigit(s.charAt(i))){
                str.append(Character.toLowerCase(s.charAt(i)));
            }
        }
        String original = str.toString();
        String reversed = str.reverse().toString();
        return original.equals(reversed);
    }
}